### ISSUE TITLE
Add new on-primary and on-secondary variables to mdc-theme

### ISSUE THREAD

**Issue by lynnjepsen** [material-components/material-components-web#2481]
`mdc-theme-on-primary` is global variable clients can set to a hex value before importing any MD Web components. Various MD Web components will look to `mdc-theme-on-primary` for color, e.g. button, and top app bar.

This will replace the "text-TEXT STYLE-on-primary". Only the "primary" text style was actually used in any MD Web components, and always had 100% opacity.

We will demonstrate this `mdc-theme-on-primary` with a new "shrine" custom theme on the theme catalog page. Shrine has a pink color for primary, and a dark brown for on-primary
![image](https://user-images.githubusercontent.com/4703432/38053994-758977c4-328a-11e8-8397-0d486fa9f492.png)

![image](https://user-images.githubusercontent.com/4703432/38053986-6f24c1d6-328a-11e8-9f0f-48a16ccd1231.png)

**Comment by englishextra** [comment]
@lynnjepsen I'm afraid this will perplex/complicate the whole procedure of tweaking MDC framwork to your custom values.

For instance - if a set primary color to somewhat of a blue color - then - yes - i would have links of the same blue color which I dont need in some cases. So - I have to write my own overirdeds to MDC which is supposed to be. Dont they? Now. you offer to make it even more complex,

To sum up - this will complicate and not ease the whole dev work based on MDC.

**Comment by kfranqueiro** [comment]
@englishextra to clarify, `on-primary` refers to a color _separate_ from `primary`, which will default to either black or white (whichever one offers better contrast when text of that color appears on top of the primary color). This is very similar to how e.g. `text-primary-on-primary` works right now.

At the same time, currently we have many `text-...-on-...` values, many of which ended up not being very useful, so we're aiming to simplify that, while still providing the same type of behavior, and additionally offering the possibility of directly customizing text colors to something other than black or white.

If you're still concerned that this will complicate matters, could you illustrate in a gist or codepen the type of use case you're concerned about?

**Comment by englishextra** [comment]
@kfranqueiro Oh it's clear now. thanks.
